query-engine/core/src/interpreter/query_interpreters/inmemory_record_processor.rs

```diff
@@ -188,3 +188,54 @@ impl InMemoryRecordProcessor {
         self.take.or(self.skip).is_some() || self.cursor.is_some()
     }
 }
+
+pub struct InMemoryRecordProcessorBuilder {
+    args: QueryArguments,
+}
```

Comment:
What is the goal of this builder? The way it is set up now, it isn't adding much value besides adding a layer of indirection.

Reply:
The builder has been removed per the call we had 👍